packages/core/source/parse.ts

```diff
@@ -86,10 +88,30 @@ Utilisez leur contrepartie française : 'oui' / 'non'`,
 
 const compiledGrammar = Grammar.fromCompiled(grammar)
 
-function parseExpression(
-	rawNode,
-	context: Context
-): Record<string, unknown> | undefined {
+/**
+ * Parse a publicodes expression into an JSON object representing the AST.
+ *
+ * The parsing is done with the [nearley](https://nearley.js.org/) parser based
+ * on the [grammar](https://github.com/betagouv/publicodes/blob/290c079d1f22baed77a96bdd834ef6cb44fa8da9/packages/core/source/grammar.ne)
+ *
+ * @param rawNode The expression to parse
+ * @param dottedName The dottedName of the rule being parsed
+ *
+ * @returns The parsing result as a JSON object
+ *
+ * @throws A `SyntaxError` if the expression is invalid
+ * @throws A `PublicodesInternalError` if the parser is unable to parse the expression
+ *
+ * @example
+ * ```ts
+ * parseExpression('20.3 * nombre', 'foo . bar')
+ * // returns { "*": [ { constant: { type: "number", nodeValue: 20.3 } }, { variable:"nombre" } ] }
+ * ```
+ */
+export function parseExpression(
+	rawNode: string,
+	dottedName: string
+): Record<string, unknown> {
```

Comment:
Suggestion pour un type plus approprié :
```suggestion
): ASTNode {
```

Reply:
Hmm je ne comprends pas trop pourquoi ce serait plus approprié, en effet, la valeur retournée par `parseExpression` ne correspond pas un AST. Elle correspond aux différentes valeurs de retours des fonctions de `grammarFunctions.js`.

Reply:
Effectivement, au temps pour moi, je me suis emmêlé les pinceaux. Le type proposé est très clair, et cela va dans le sens d'une meilleure spécification des structures de données intermédiaire. 

Dernière petite suggestion : serait-il possible de déplacer la fonction et ses types dans un fichier séparé pour d'avantage de clarté ?

Sinon tout est bon pour moi !